src/components/locations/AddLocation.tsx

```diff
@@ -36,26 +47,37 @@ export function AddLocation({
     rightPosition !== 0 && closeSearchPanel();
   };
 
-  if (isError) return <Error />;
-
   return (
     <Container
       $rightPosition={rightPosition}
       onTransitionEnd={onTransitionEndHandler}
     >
       <SearchBar placeholder="동명(읍, 면)으로 검색 (ex. 서초동)" />
-      {isLoading ? (
-        <Loader />
+      {isError ? (
+        <Error />
       ) : (
         <Content>
-          {data.map(location => (
-            <LocationItem
-              key={location.id}
-              onClick={() => onClickLocationItem(location.item)}
-            >
-              {location.item}
-            </LocationItem>
-          ))}
+          {data?.pages.map(page => {
+            return page.locations.map((location, index) =>
+              index === page.locations.length - 1 ? (
+                <LocationItem
+                  ref={lastItemRef}
+                  key={location.id}
+                  onClick={() => onClickLocationItem(location.name)}
+                >
+                  {location.name}
+                </LocationItem>
+              ) : (
+                <LocationItem
+                  key={location.id}
+                  onClick={() => onClickLocationItem(location.name)}
+                >
+                  {location.name}
+                </LocationItem>
+              )
+            );
+          })}
```

Comment:
```suggestion
 {data?.pages.map(page => {
    return page.locations.map((location, index) => {
        const isLastItem = index === page.locations.length - 1;
        
        return (
            <LocationItem
                ref={isLastItem ? lastItemRef : null}
                key={location.id}
                onClick={() => onClickLocationItem(location.name)}
            >
                {location.name}
            </LocationItem>
        );
    });
})}
```

이렇게 사용하면 중복을 줄일 수 있을 것 같네요.

Reply:
훨씬 보기 좋네요. 바로 한번 반영해 보겠습니다